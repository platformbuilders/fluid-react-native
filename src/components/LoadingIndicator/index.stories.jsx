import { storiesOf } from '@storybook/react-native';
import loadingIndicatorMarkdown from './loadingIndicator.md';

storiesOf('Loading Indicator', module)
  .addParameters({
    readme: {
      content: loadingIndicatorMarkdown,
    },
    viewport: {
      defaultViewport: '',
    },
  })
  .add('Example', () => {});
