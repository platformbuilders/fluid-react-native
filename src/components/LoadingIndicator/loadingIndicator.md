
# Loading Indicator 🔁 

## Example
![ezgif com-video-to-gif](https://user-images.githubusercontent.com/44801113/81099998-2869d900-8ee2-11ea-9b81-f3df2a34bcd9.gif)


## Source Code
```js
import React from 'react';
import { LoadingIndicator } from '@platformbuilders/react-native-ui';

const Home: React.FC = () => {
  return (
      <LoadingIndicator large />
  );
}

export default Home;
```

## Props 

| Name  | Required | Type | Default Value | Description 
| ------------- | ------------- | ------------- |------------- |------------- |
| animation | false | string | circular | options: circular, linear, button  |
| large | false | boolean | false | size loading  |

