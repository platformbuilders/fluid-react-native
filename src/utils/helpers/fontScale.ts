import { RFValue } from 'react-native-responsive-fontsize';

export const scaledFontSize = (
  size: number | string,
  baselineFactor?: 280,
): number => {
  const value = typeof size === 'string' ? parseInt(size, 10) : size;
  return RFValue(value, baselineFactor);
};
