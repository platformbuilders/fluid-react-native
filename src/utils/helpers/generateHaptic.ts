import Haptic, { HapticFeedbackType } from 'react-native-haptic';
import { isIOS } from './isIOS';

export const generateHaptic = (haptic: HapticFeedbackType = 'impact'): void => {
  if (isIOS()) {
    Haptic.generate(haptic);
  }
};
