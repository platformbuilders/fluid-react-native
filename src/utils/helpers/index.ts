export {
  currencyParser,
  parseToThousands,
  removeWhiteSpaces,
  toOnlyNumbers,
  getTheme,
  ifStyle,
  switchStyle,
} from '@platformbuilders/helpers';
export * from './generateHaptic';
export * from './getByTypographyVariant';
export * from './fontScale';
export * from './shadow';
