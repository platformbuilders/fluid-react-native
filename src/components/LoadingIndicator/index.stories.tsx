import type { Meta } from '@storybook/react';
import { VARIANT_TYPE } from '../../enums/size.enum';
import LoadingIndicator from '.';

const meta: Meta<typeof LoadingIndicator> = {
  title: 'LoadingIndicator',
  component: LoadingIndicator,
  argTypes: {
    variant: { control: { type: 'select' }, options: VARIANT_TYPE },
  },
  args: {
    accessibility: 'LoadingIndicator',
    contrast: false,
    large: false,
    variant: 'linear',
  },
  parameters: {
    notes:
      'Esse componente tem como função auxiliar na manipulação, manutenção e padronização de indicadores de carregamento dentro de uma aplicação.',
  },
};

export default meta;

export const Default = {};
