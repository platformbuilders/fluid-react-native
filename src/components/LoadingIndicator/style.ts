import styled from 'styled-components/native';
import Animation from 'lottie-react-native';
import { moderateScale } from 'react-native-size-matters';
import { getTheme } from '../../utils/helpers';
import { AnimationObject, LoadingVariants } from '../../types';
import LoadingAnimation from '../../assets/animations/loading.json';

export const circularLoading = getTheme('circularLoading');
export const contrastCircularLoading = getTheme('contrastCircularLoading');
export const buttonLoading = getTheme('buttonLoading');
export const contrastButtonLoading = getTheme('contrastButtonLoading');
export const linearLoading = getTheme('linearLoading');
export const contrastLinearLoading = getTheme('contrastLinearLoading');

type IndicatorProps = {
  contrast: boolean;
  variant: LoadingVariants;
  testID: string;
  accessibilityLabel: string;
};

const loadingVariant = (
  props: any,
): string | AnimationObject | { uri: string } => {
  const { variant, contrast } = props;
  switch (variant) {
    case 'button':
      return contrast
        ? contrastButtonLoading(props)
        : buttonLoading(props) || LoadingAnimation;
    case 'circular':
      return contrast
        ? contrastCircularLoading(props)
        : circularLoading(props) || LoadingAnimation;
    case 'linear':
      return contrast
        ? contrastLinearLoading(props)
        : linearLoading(props) || LoadingAnimation;
    default:
      return contrast ? contrastCircularLoading(props) : circularLoading(props);
  }
};

export const smallSize = {
  width: moderateScale(60),
  height: moderateScale(60),
};

export const largeSize = {
  width: moderateScale(120),
  height: moderateScale(120),
};

export const Indicator = styled(Animation).attrs((props: any) => ({
  source: loadingVariant(props),
  autoPlay: true,
  loop: true,
}))<IndicatorProps>``;
