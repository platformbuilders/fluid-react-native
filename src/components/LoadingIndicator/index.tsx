import React, { FC } from 'react';
import { LoadingType } from '../../types';
import { Indicator, smallSize, largeSize } from './style';

const Loading: FC<LoadingType> = ({
  large = false,
  contrast = false,
  variant = 'circular',
  accessibility,
  ...rest
}) => (
  <Indicator
    testID={accessibility || 'loading'}
    accessibilityLabel={accessibility || 'Aguarde'}
    variant={variant}
    contrast={contrast}
    style={large ? largeSize : smallSize}
    {...rest}
  />
);

export default Loading;
