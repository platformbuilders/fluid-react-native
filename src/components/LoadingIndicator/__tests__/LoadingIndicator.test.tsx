import 'jest';
import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components/native';
import LoadingIndicator from '..';
import { theme } from '../../../test/helpers';

describe('<LoadingIndicator />', () => {
  it('should render LoadingIndicator correctly', () => {
    const render = renderer.create(
      <ThemeProvider theme={theme}>
        <LoadingIndicator />
      </ThemeProvider>,
    );

    expect(render.toJSON()).toMatchSnapshot();
  });
});
