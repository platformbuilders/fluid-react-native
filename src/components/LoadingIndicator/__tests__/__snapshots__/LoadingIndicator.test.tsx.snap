// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingIndicator /> should render LoadingIndicator correctly 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
      Array [
        Object {},
        Object {
          "height": 94.28571428571428,
          "width": 94.28571428571428,
        },
      ],
    ]
  }
>
  <View
    accessibilityLabel="Aguarde"
    autoPlay={true}
    contrast={false}
    enableMergePathsAndroidForKitKatAndAbove={false}
    forwardedRef={[Function]}
    loop={true}
    onAnimationFinish={[Function]}
    onLayout={[Function]}
    progress={0}
    resizeMode="contain"
    speed={1}
    style={
      Object {
        "height": 94.28571428571428,
        "width": 94.28571428571428,
      }
    }
    testID="loading"
    variant="circular"
  />
</View>
`;

exports[`<LoadingIndicator /> should render LoadingIndicator with accessibility 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
      Array [
        Object {},
        Object {
          "height": 94.28571428571428,
          "width": 94.28571428571428,
        },
      ],
    ]
  }
>
  <View
    accessibilityLabel="Aguarde"
    autoPlay={true}
    contrast={false}
    enableMergePathsAndroidForKitKatAndAbove={false}
    forwardedRef={[Function]}
    loop={true}
    onAnimationFinish={[Function]}
    onLayout={[Function]}
    progress={0}
    resizeMode="contain"
    speed={1}
    style={
      Object {
        "height": 94.28571428571428,
        "width": 94.28571428571428,
      }
    }
    testID="loading"
    variant="circular"
  />
</View>
`;

exports[`<LoadingIndicator /> should render LoadingIndicator with contrast 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
      Array [
        Object {},
        Object {
          "height": 94.28571428571428,
          "width": 94.28571428571428,
        },
      ],
    ]
  }
>
  <View
    accessibilityLabel="Aguarde"
    autoPlay={true}
    contrast={true}
    enableMergePathsAndroidForKitKatAndAbove={false}
    forwardedRef={[Function]}
    loop={true}
    onAnimationFinish={[Function]}
    onLayout={[Function]}
    progress={0}
    resizeMode="contain"
    speed={1}
    style={
      Object {
        "height": 94.28571428571428,
        "width": 94.28571428571428,
      }
    }
    testID="loading"
    variant="circular"
  />
</View>
`;

exports[`<LoadingIndicator /> should render LoadingIndicator with large 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
      Array [
        Object {},
        Object {
          "height": 188.57142857142856,
          "width": 188.57142857142856,
        },
      ],
    ]
  }
>
  <View
    accessibilityLabel="Aguarde"
    autoPlay={true}
    contrast={false}
    enableMergePathsAndroidForKitKatAndAbove={false}
    forwardedRef={[Function]}
    loop={true}
    onAnimationFinish={[Function]}
    onLayout={[Function]}
    progress={0}
    resizeMode="contain"
    speed={1}
    style={
      Object {
        "height": 188.57142857142856,
        "width": 188.57142857142856,
      }
    }
    testID="loading"
    variant="circular"
  />
</View>
`;

exports[`<LoadingIndicator /> should render LoadingIndicator with variant 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
      Array [
        Object {},
        Object {
          "height": 94.28571428571428,
          "width": 94.28571428571428,
        },
      ],
    ]
  }
>
  <View
    accessibilityLabel="Aguarde"
    autoPlay={true}
    contrast={false}
    enableMergePathsAndroidForKitKatAndAbove={false}
    forwardedRef={[Function]}
    loop={true}
    onAnimationFinish={[Function]}
    onLayout={[Function]}
    progress={0}
    resizeMode="contain"
    speed={1}
    style={
      Object {
        "height": 94.28571428571428,
        "width": 94.28571428571428,
      }
    }
    testID="loading"
    variant="linear"
  />
</View>
`;
